subsys/jaxrs/src/main/java/org/commonjava/indy/bind/jaxrs/util/ResponseHelper.java

```diff
@@ -175,39 +143,46 @@ public ResponseBuilder setInfoHeaders( final ResponseBuilder builder, final Tran
         boolean lastModSet = false;
         boolean lenSet = false;
         boolean conTypeSet = false;
-    
-        
 
         if ( exchangeMetadata != null )
         {
             for ( final Map.Entry<String, List<String>> headerSet : exchangeMetadata.getResponseHeaders().entrySet() )
             {
                 final String key = headerSet.getKey();
-                if ( ApplicationHeader.content_type.upperKey().equals( key ) )
-                {
-                    logger.debug( "Marking {} as already set.", ApplicationHeader.content_type.upperKey() );
-                    conTypeSet = true;
-                }
-                else if ( ApplicationHeader.last_modified.upperKey().equals( key ) )
-                {
-                    logger.debug( "Marking {} as already set.", ApplicationHeader.last_modified.upperKey() );
-                    lastModSet = true;
-                }
-                else if ( ApplicationHeader.content_length.upperKey().equals( key ) )
+                final List<String> val = headerSet.getValue();
+                if ( val != null && !val.isEmpty() )
                 {
-                    logger.debug( "Marking {} as already set.", ApplicationHeader.content_length.upperKey() );
-                    lenSet = true;
-                    if ( !includeContentLength )
+                    for ( final String value : headerSet.getValue() )
                     {
-                        continue;
+                        if ( StringUtils.isBlank( value ) )
+                        {
+                            logger.warn( "Warning: Http metadata contains a blank value for key {}, will skip setting it.", ApplicationHeader.content_type.upperKey() );
```

Comment:
should be this? logger.warn( "Warning: Http metadata contains a blank value for key {}, will skip setting it.", key )

Reply:
Oops, a copy paste error. Will fix.